### ISSUE TITLE
Handle lax static typing similarly to readonly

### ISSUE THREAD

**Issue by jclark** [ballerina-platform/ballerina-spec#488]
At the moment, lax static typing is not well integrated into the type system: laxness is not part of the type. This makes it hard to make lax static typing work coherently. There's also no way for users to say whether a type should be laxly typed (this is #198).

The concept of lax static typing is that it is a way to achieve the effect of dynamic typing. The programmer has in mind a particular static type, but has not made it explicit in the program; with a lax type, an expression that would be rejected at compile-time is allowed at compile-time and instead is allowed to produce an error value at run-time.

C# provides this [feature](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/types/using-type-dynamic) using a type `dynamic`.

One idea is to approach this similarly to `readonly`.  We could then define `json` as:

```
type json = dynamic & 
  (()|boolean|int|float|decimal|string|json[]|map<json>)
```

In other words, we are saying laxness becomes a non-primary aspect of type. It is almost like saying values have a dynamic bit, just like they have a read-only bit. But not quite: this should be a compile-time feature. When you do a cast `<dynamic>`, instead of checking that the value actually has a dynamic bit set, the cast would always succeed. The static type checking would though work exactly as if values had this dynamic bit.

This could potentially provide a basis on which to solve #487.
